### DISCUSSION TITLE
TS & TSX Language Server issues

### DISCUSSION THREAD

**Discussion by rglasgow-maxpreps** [zed-industries/zed#7004] [Feature Requests]
Hello, hopefully, someone can help resolve this issue.

I have an issue getting the TS & TSX language servers to work. When I initially open a TS / TSX file, I can see that the language server is being downloaded for that language. However, it spends a number of cycles going from "Downloading Language server" to "Checking for updates." Once the cycle resolves, it looks good to go without error, except it's not. I get no autocompletion at all. 

I have tested GO and Rust, and they appear to work without issue. Python is having the same issue as TSX and TS. 

It's worth mentioning all of these languages work perfectly fine on my other computer running the same build and settings (default)

**Comment by aurindo10** [comment]
I have the same issue. I just can't use because of that and there is nothing to help on internet

**Reply by elcharitas** [reply]
TSX should work with no issues whatsoever (try restarting zed). However, for python, you would need to configure pyright which is the LSP zed uses. More information can be found here - https://github.com/zed-industries/zed/blob/main/docs/src/languages/python.md

**Reply by rglasgow-maxpreps** [reply]
I figured it out.

@aurindo10, you need to first delete the fish files found in the zed config folder.
Find your global node_modules and delete the LSP file in there. Those two things solved my issue.

**Comment by elcharitas** [comment]
TSX should work with no issues whatsoever (try restarting zed). However, for python, you would need to configure pyright which is the LSP zed uses. More information can be found here - https://github.com/zed-industries/zed/blob/main/docs/src/languages/python.md

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
